nuxeo-core/nuxeo-core-api/src/main/java/org/nuxeo/ecm/core/api/ScrollResultImpl.java

```diff
@@ -24,30 +24,21 @@
 /**
  * @since 8.4
  */
-public class ScrollResultImpl<T> implements ScrollResult {
+public class ScrollResultImpl<T> implements ScrollResult<T> {
 
-    protected static final ScrollResult EMPTY_RESULT = new ScrollResultImpl("empty", Collections.emptyList());
+    protected static final ScrollResult EMPTY_RESULT = new ScrollResultImpl<>("empty", Collections.emptyList());
 
     protected final String scrollId;
 
-    protected final List<String> ids;
-
     protected final List<T> results;
 
-    public static ScrollResult emptyResult() {
+    @SuppressWarnings("unchecked")
+    public static <O> ScrollResult<O> emptyResult() {
         return EMPTY_RESULT;
```

Comment:
`return (<O>) EMPTY_RESULT;` I guess

Reply:
Indeed, this wasn't necessary because I forgot the generic type on the constant 👍